### PR TITLE
PP-737: expect failure causes a test error rather than failure

### PR DESCRIPTION
**Pull request by hirenvadalia**
#### Issue-ID

* **[PP-737](https://pbspro.atlassian.net/browse/PP-737)**

#### Problem description
* expect failure should be reported as failure rather than error

#### Cause / Analysis
* currently expect failure is reported as error where it is not because expect fail is something like tests desired condition doesn't meet which is an test failure not an error

#### Solution description
* modified PtlExpectError to new introduced PtlfailureException (which is nothing but PTL modified version of assert failure aka failure in terms of testing framework) from PtlException (which is nothing but PTL modified version of Exception aka error in terms of testing framework)

#### Checklist:

- [x] I have joined the **[pbspro community forum](http://community.pbspro.org/)**.
- [x] My pull request contains a **single, signed** commit. See **[setting up gpg signature](https://pbspro.atlassian.net/wiki/display/DG/Signing+Your+Git+Commits)**.
- [x] My code follows the **[coding style](https://pbspro.atlassian.net/wiki/display/DG/Coding+Standards)** of this project.
- [ ] My change requires project documentation. See **[required documentation checklist](https://pbspro.atlassian.net/wiki/display/DG/Checklist+for+Developing+Features+and+Bug+Fixes)** for details.
- [ ] I have added documentation in the **[project documentation area](https://pbspro.atlassian.net/wiki/display/PD)**.
- [ ] I have added new **PTL test(s) to my commit**. (See **[using PTL for testing](https://pbspro.atlassian.net/wiki/display/DG/Using+PTL+for+Testing)**) *(or)*
- [ ] I have added  **manual test(s) to the Jira ticket and explained why PTL is not appropriate** for this case.
- [x] All new and existing automated tests have passed. (See **[running automated PTL tests](https://pbspro.atlassian.net/wiki/display/DG/PTL+Quick+Start+Guide)**).
- [x] I have attached **test logs to the Jira ticket** as evidence of testing/verification.

__***For further information please visit the [Developer Guide Home](https://pbspro.atlassian.net/wiki/display/DG/Developer+Guide+Home).***__

### CODE REVIEW

File: test/fw/ptl/lib/pbs_testlib.py [hunk 1]
```diff
@@ -315,6 +315,42 @@ def __repr__(self):
...
+
+    :param rv: Return value set for the failure occured during PTL
+               operation
+    :type rv: int or None.
+    :param rc: Return code set for the failure occured during PTL
+               operation
+    :type rc: int or None.
+    :param msg: Message set for the failure occured during PTL operation
+    :type msg: str or None.
+    :param post: Execute necessary cleanup if not None
```
Comment (arungrover) [inline test/fw/ptl/lib/pbs_testlib.py]: what is the type of "post"?

Comment (arungrover) [inline test/fw/ptl/lib/pbs_testlib.py]: what is the type of post here?

Comment (agrawalravi90) [inline test/fw/ptl/lib/pbs_testlib.py]: post is a callback function right? could you please explicitly mention that along with the prototype of the callback function?

File: test/fw/ptl/lib/pbs_testlib.py [hunk 2]
```diff
@@ -315,6 +315,42 @@ def __repr__(self):
                 str(self.rv) + ', msg=' + str(self.msg) + ')')
 
 
+class PtlfailureException(AssertionError):
```
Comment (agrawalravi90) [inline test/fw/ptl/lib/pbs_testlib.py]: shouldn't 'f' of 'failure' be caps?

**Comment by bhroam** [comment]
What is the status of this PR?  It's been lingering for quite some time.

File: test/fw/ptl/lib/pbs_testlib.py (hunk 1 above)
Comment (hirenvadalia) [inline test/fw/ptl/lib/pbs_testlib.py]: Added type of post.

Comment (hirenvadalia) [inline test/fw/ptl/lib/pbs_testlib.py]: Yes, modified doc for same. But regarding prototype of callback function, its free form, there is no defined prototype for it. You can choose any number of argument with any type of argument to that callback function and which I don't think we need to specify that here.

File: test/fw/ptl/lib/pbs_testlib.py (hunk 2 above)
Comment (hirenvadalia) [inline test/fw/ptl/lib/pbs_testlib.py]: done, made 'f' in caps.

**Comment by hirenvadalia** [comment]
@arungrover Added pre and post fix test results to ticket. Please have a look and let me know if anything is missing.

**Comment by arungrover** [comment]
@hirenvadalia merging it in. Please add fix version to the ticket.

**Comment by hirenvadalia** [comment]
@arungrover Done, added 17.1.1 as fix version in ticket.

**Pull request merged**